test/functional/userAuth.test.js

```diff
@@ -40,6 +45,40 @@ describe('Authentication', () => {
     })
   })
 
+  describe('when logging in', () => {
+    before(() => sinon.stub(process.env, 'APPROVED_DOMAINS').value('(.* )?st.com'))
+    after(() => sinon.restore())
+
+    it('should check for regex domains', () => {
+      return request(app)
+        .get('/')
+        .expect(200)
+    })
+
+    before(() => sinon.stub(process.env, 'APPROVED_DOMAINS').value('test.user@test.com'))
```

Comment:
[Mocha's documentation](https://mochajs.org/#hooks) seems to indicate that all `before` and `after` hooks run at the start end end of the block they are in, not synchronously with the tests, top to bottom. 

If there are still unique before/after hooks that need to be run for each of these scenarios, maybe we could create additional `describe` blocks to scope these scenarios? That would follow the convention used elsewhere within our tests.